zero-loads-app/src/loads/stub/sensor/definitions/units.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass
+from typing import Annotated, Any, Dict, TypeAlias, get_args, get_origin, get_type_hints
+
+from generator import Generator, GeneratorFactory
+from pydantic import (
+    AliasGenerator,
+    BaseModel,
+    ConfigDict,
+    Field,
+)
+
+from .util import hyphenize
+
+TOPIC_PREFIX: str = "sail-systems/"
+
+
+@dataclass(frozen=True)
+class Unit:
+    unit: str
+
+
+Position: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="mm")]
+RelativePosition: TypeAlias = Annotated[float, Field(ge=0, lt=1), Unit(unit="%")]
+Load: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="ton")]
+Torque: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="Nm")]
+RotationalSpeed: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="rpm")]
+Temperature: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="°C")]
+
+
+class ComponentMeta(BaseModel):
+    topic: str
+
+
+def component_meta(*args, **kwargs):
+    return Field(json_schema_extra=ComponentMeta(*args, **kwargs).model_dump())
+
+
+class LoadsModel(BaseModel):
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(
+            serialization_alias=hyphenize,
+        )
+    )
+
+    @classmethod
+    def gen_config(cls, interval: int = 10) -> list[dict]:
+        """Generate configuration for data generation."""
+        hints = get_type_hints(cls, include_extras=True)
+        config = []
+
+        for component, value_definition in hints.items():
+            base_type, meta = cls._unwrap_annotated(value_definition)
+
+            topic = TOPIC_PREFIX + cls._extract_topic(meta, component)
```

Comment:
Prefer f-strings for strings (even if it's concatenation). Then you can remove the trailing `/` from the prefix and put it in here